python/cuml/comm/serialize.py

```diff
@@ -0,0 +1,58 @@
+#
+# Copyright (c) 2020, NVIDIA CORPORATION.
```

Comment:
Would it make more sense for this to live in `python/cuml/dask/comm` since it's specifically for Dask serializers?

Reply:
it is about serializing cuml array mainly which lives in the base folders, and then its consistent with cuDF. Additionally, dask already has a dependency on this location, without this PR dask master fails in our CI: 

```
E         File "/home/galahad/miniconda3/envs/ns0218/lib/python3.7/site-packages/distributed/protocol/__init__.py", line 102, in _register_cuml
E           from cuml.comm import serialize
E       ImportError: cannot import name 'serialize' from 'cuml.comm' (unknown location)
```

Reply:
I could go either way with it, really. It just seems more natural to have it in the `cuml.dask` package since it's specifically for Dask. I can remove my request changes, I just wanted to make sure I got a chance to ask the question before this was merged.

Reply:
@jakirkham thoughts? I'm fine with it living in either place as well

Reply:
We can change the location. Was just [mirroring cuDF]( https://github.com/rapidsai/cudf/blob/89bb875ec8d0bd6e72a18ac5706c748626066714/python/cudf/cudf/comm/serialize.py ).

Interesting. My understanding was (and past experience has been) this would be a no-op if the `import` can't be loaded. Could you please share a bit more context @dantegd? 🙂

Reply:
Locally without the changes of this PR, when running pytests kmeans ran into the issue I posted above

Reply:
Same in CI of PR #1699: 
```
cuml/test/dask/test_kmeans.py::test_end_to_end[None-5-10-1000.0] distributed.protocol.core - CRITICAL - Failed to deserialize

Traceback (most recent call last):

  File "/conda/envs/gdf/lib/python3.6/site-packages/distributed/protocol/core.py", line 124, in loads

    value = _deserialize(head, fs, deserializers=deserializers)

  File "/conda/envs/gdf/lib/python3.6/site-packages/distributed/protocol/serialize.py", line 268, in deserialize

    return loads(header, frames)

  File "/conda/envs/gdf/lib/python3.6/site-packages/distributed/protocol/serialize.py", line 80, in serialization_error_loads

    raise TypeError(msg)

TypeError: Could not serialize object of type KMeansMG.

Traceback (most recent call last):

  File "/conda/envs/gdf/lib/python3.6/site-packages/distributed/protocol/serialize.py", line 191, in serialize

    header, frames = dumps(x, context=context) if wants_context else dumps(x)

  File "/conda/envs/gdf/lib/python3.6/site-packages/distributed/protocol/serialize.py", line 37, in dask_dumps

    dumps = dask_serialize.dispatch(type(x))

  File "/conda/envs/gdf/lib/python3.6/site-packages/dask/utils.py", line 491, in dispatch

    register()

  File "/conda/envs/gdf/lib/python3.6/site-packages/distributed/protocol/__init__.py", line 102, in _register_cuml

    from cuml.comm import serialize

ModuleNotFoundError: No module named 'cuml.comm'


FAILED
``` 
of log https://gpuci.gpuopenanalytics.com/blue/organizations/jenkins/rapidsai%2Fgpuci%2Fcuml%2Fprb%2Fcuml-gpu-build/detail/cuml-gpu-build/4988/pipeline

Reply:
Thanks for the context. It seems it will raise if `cuml` is present, but `cuml.comm` is not present. PR ( https://github.com/dask/distributed/pull/3499 ) should fix this.

Reply:
So the CI issue should be fixed. Please let me know if we should move the location and can update accordingly

Reply:
Both dask and non-dask locations seem reasonable, so I'm partial to keeping consistent with cudf as it is now.